### PR TITLE
xmake: improve install scripts

### PR DESCRIPTION
**Pull request by waruqi**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/xmake.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class Xmake < Formula
   desc "A cross-platform build utility based on Lua"
   homepage "https://xmake.io/"
-  url "https://github.com/xmake-io/xmake/releases/download/v2.2.9/xmake-v2.2.9.tar.gz"
+  url "https://cdn.jsdelivr.net/gh/xmake-mirror/xmake-releases@2.2.9/xmake-v2.2.9.tar.gz"
```
Comment (SMillerDev) [inline Formula/xmake.rb]: Please use the original upstream instead of a mirror

**Review by SMillerDev — COMMENTED** [review]
Please don't change the url

File: Formula/xmake.rb (hunk 1 above)
Comment (waruqi) [reply Formula/xmake.rb]: ok, I will restore it. 

But it is difficult to access github and bintray fastly in China. Are there any other good other good solutions?

Comment (waruqi) [reply Formula/xmake.rb]: I have fixed url.

**Comment by jonchang** [comment]
Does this need a `revision` bump?

**Comment by waruqi** [comment]
> Does this need a `revision` bump?

What difference does it add a revision? How should I add it?

**Comment by jonchang** [comment]
Will users need to install new binaries for this formula when they run `brew upgrade`? Or is this merely a build system change on our end?

**Comment by waruqi** [comment]
> Will users need to install new binaries for this formula when they run `brew upgrade`? Or is this merely a build system change on our end?

Don't need it，only improve the build script.

**Comment by jonchang** [comment]
Thanks for contributing to Homebrew, @waruqi!

**Comment by waruqi** [comment]
Thanks!

**Pull request merged**